pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdConnectors.java

```diff
@@ -0,0 +1,612 @@
+     */
+    @Getter
+    abstract class BaseCommand extends CliCommand {
+        @Override
+        void run() throws Exception {
+            processArguments();
+            runCmd();
+        }
+
+        void processArguments() throws Exception {
+        }
+
+        abstract void runCmd() throws Exception;
+    }
+
+    @Parameters(commandDescription = "Run the Pulsar Function locally (rather than deploying it to the Pulsar cluster)")
+
+    class LocalSourceRunner extends CreateSource {
+
+        @Parameter(names = "--brokerServiceUrl", description = "The URL for the Pulsar broker")
+        protected String brokerServiceUrl;
+
+        @Override
+        void runCmd() throws Exception {
+            CmdFunctions.startLocalRun(createSourceConfigProto2(sourceConfig),
+                    sourceConfig.getParallelism(), brokerServiceUrl, jarFile, admin);
+        }
+    }
+
+    class LocalSinkRunner extends CreateSink {
+
+        @Parameter(names = "--brokerServiceUrl", description = "The URL for the Pulsar broker")
+        protected String brokerServiceUrl;
+
+        @Override
+        void runCmd() throws Exception {
+            CmdFunctions.startLocalRun(createSinkConfigProto2(sinkConfig),
+                    sinkConfig.getParallelism(), brokerServiceUrl, jarFile, admin);
+        }
+    }
+
+    @Parameters(commandDescription = "Create Pulsar source connectors")
+    class CreateSource extends BaseCommand {
+        @Parameter(names = "--fqfn", description = "The Fully Qualified Function Name (FQFN) for the function")
+        protected String fqfn;
+        @Parameter(names = "--tenant", description = "The function's tenant")
+        protected String tenant;
+        @Parameter(names = "--namespace", description = "The function's namespace")
+        protected String namespace;
+        @Parameter(names = "--name", description = "The function's name")
+        protected String name;
+        @Parameter(names = "--processingGuarantees", description = "The processing guarantees (aka delivery semantics) applied to the Source")
+        protected FunctionConfig.ProcessingGuarantees processingGuarantees;
+        @Parameter(names = "--className", description = "The function's class name")
+        protected String className;
+        @Parameter(names = "--topicName", description = "Pulsar topic to ingress data to")
+        protected String topicName;
+        @Parameter(names = "--serdeClassName", description = "")
+        protected String serdeClassName;
+        @Parameter(names = "--parallelism", description = "")
+        protected String parallelism;
+        @Parameter(
+                names = "--jar",
+                description = "Path to the jar file for the function (if the function is written in Java)",
+                listConverter = StringConverter.class)
+        protected String jarFile;
+
+        @Parameter(names = "--functionConfigFile", description = "The path to a YAML config file specifying the "
```

Comment:
Isn't this leaking the abstraction?

Reply:
call it connectorConfigFile?

Reply:
yes. And this should take sinkConfig or sourceConfig